orcid-core/src/main/java/org/orcid/core/manager/impl/OrcidProfileManagerImpl.java

```diff
@@ -423,10 +423,8 @@ private void addSourceToWorks(OrcidProfile orcidProfile, String amenderOrcid) {
     private void addSourceToWorks(OrcidWorks orcidWorks, String amenderOrcid) {
         if (orcidWorks != null && !orcidWorks.getOrcidWork().isEmpty() && amenderOrcid != null) {
             for (OrcidWork orcidWork : orcidWorks.getOrcidWork()) {
-                if (orcidWork.getSource() == null || StringUtils.isBlank(orcidWork.getSource().retrieveSourcePath())) {
```

Comment:
@rcpeters The source value that we  put in the XML may be valid, but it cannot be any other than the client who is trying to call the API. So it doesn't matter if the source is valid or not. It has to be overwritten with the client ID.
There may be a case that the XML already contains the client id and we try to overwrite it again by the same value, but that doesnt increase the complexity of the code.